src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+                // checking the returned content for the string 'ERROR'. It will retry up to 5 times, with a 10 second delay.
+                if (line.contains("ERROR") ) {
+                    retryCount++;
+                    if (retryCount < 5) {
+                        Thread.sleep(10000);
+                        logger.warn("Biomart query failed. Trying again...");
+                        return queryBiomart(queryString, retryCount);
+                    }
+                    throw new Exception("Biomart query failed with message: " + line);
+                } else {
+                    biomartResponseLines.add(line);
+                }
+            }
+            return biomartResponseLines;
+
+        } else {
+            retryCount++;
+            if (retryCount < 5) {
+                Thread.sleep(10000);
+                logger.warn("Unable to connect to Biomart. Trying again...");
+                return queryBiomart(queryString, retryCount);
+            }
+            throw new Exception("Unable to connect to Biomart with URL: " + queryString);
+        }
+    }
+
+    /**
+     * Function that takes species name attribute from config file (eg: Homo sapiens) and modifies it to
+     * match Biomart formatting (first letter from primary species name + secondary name, all lowercase -- eg: hsapiens).
+     * @return List<String> of species names in Biomart format (eg: hsapiens).
+     * @throws IOException
+     * @throws ParseException
+     */
+    private List<String> getBiomartSpeciesNames() throws IOException, ParseException {
+        // Read properties file.
+        Properties applicationProps = new Properties();
+        String propertiesLocation = System.getProperty("config.location");
+        try(FileInputStream fis = new FileInputStream(propertiesLocation))
+        {
+            applicationProps.load(fis);
+        }
+        // Get species JSON config file location, import as JSON object.
+        String pathToSpeciesConfig = applicationProps.getProperty("pathToSpeciesConfig");
+        JSONParser parser = new JSONParser();
+        Object obj = parser.parse(new FileReader(pathToSpeciesConfig));
+        JSONObject jsonObject = (JSONObject) obj;
+
+        // For each species in config file, retrieve name, modify it and then add to list.
+        List<String> biomartNames = new ArrayList<>();
+        for (Object speciesKey : jsonObject.keySet()) {
+            JSONObject speciesJson = (JSONObject) jsonObject.get(speciesKey);
+            JSONArray speciesNames = (JSONArray) speciesJson.get("name");
+            String speciesName = (String) speciesNames.get(0);
+            String biomartName = speciesName.substring(0,1).toLowerCase() + speciesName.split(" ")[1];
+            biomartNames.add(biomartName);
+        }
+        return biomartNames;
+    }
+
+    // Gets the filepath to biomart XML file, used to build biomart query.
+    private String getBiomartXMLPath() throws IOException {
```

Comment:
Small quibble: the phrase "XMLPath" Could get confused with "XPath" which is a completely different technology. I'd recommend changing the name of the method to `getBiomartXMLFilePath` so that it's clear you're getting a file, not a path _within_ an XML document.